packages/style-engine/class-wp-style-engine.php

```diff
@@ -523,3 +610,32 @@ function wp_style_engine_get_block_supports_styles( $block_styles, $options = ar
 	}
 	return null;
 }
+
+/**
+ * Global public interface method to WP_Style_Engine->get_block_supports_styles to generate block styles from a single block style object
+ * and then, via the `enqueue` flag, will enqueue them for rendering on the frontend in a block-supports inline style tag.
+ *
+ * @access public
+ *
+ * @param string $selector     A CSS selector.
+ * @param array  $block_styles The value of a block's attributes.style.
+ *
+ * @return array<string>|null array(
+ *     'styles'        => (string) A CSS ruleset or declarations block formatted to be placed in an HTML `style` attribute or tag.
+ *     'declarations'  => (array) An array of property/value pairs representing parsed CSS declarations.
+ *     'classnames'    => (string) Classnames separated by a space.
+ * );
+ */
+function wp_style_engine_enqueue_block_supports_styles( $selector, $block_styles ) {
```

Comment:
I like the additional public method for this, as it makes it a bit clearer than adding additional options to the `wp_style_engine_get_block_supports_styles` function. That said, given we have an `$options` param on the other function, should we try to somehow consolidate the two, or do you think it's better having the two separate public functions for better clarity?

I suppose my idle thought is: is it better to have a single public function with lots of options, or separate functions that are simpler to work with, but we then have more public functions in the global namespace 🤔

Reply:
> is it better to have a single public function with lots of options, or separate functions that are simpler to work with, but we then have more public functions in the global namespace

This is the struggle. 😄 

The reason I went with two was to separate the concerns (at least ostensibly) and provide a declarative way of enqueuing block styles so the consumer wouldn't have to worry about the underlying implementation. 

That `wp_style_engine_get_block_supports_styles` accepts the enqueue option does mix concerns in that way, and I think it'd be cleaner to somehow separate parsing from enqueuing.

I'm not sure. I'll try some things out.

Reply:
So I might have gone over the top a bit, but I was trying to separate out parsing methods from compilation (to CSS strings etc)

Just testing things out.

Wondering there's a case for more abstraction, e.g., pulling out parsing methods from the main class.